bpf/bpf_lxc.c

```diff
@@ -803,6 +804,15 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx,
 		if (!info)
 			goto skip_egress_gateway;
 
+		/* If the packet is a reply, it means that outside has initiated
+		 * the connection, and so we should skip egress gateway, since
+		 * an egress policy is only matching connections originating
+		 * from a pod.
+		 */
+		if (!ct_is_reply4(get_ct_map4(&tuple), ctx, ETH_HLEN + ipv4_hdrlen(ip4), &tuple,
+				  &is_reply) && is_reply)
```

Comment:
Ah, I think we can :+1: 

I got fooled by the `CT_EGRESS` param passed to the `ct_lookup4()` call above, but I just noticed that this will set `tuple->flags` to `TUPLE_F_IN` and so we'll end up with the same key set by `ct_is_reply4()`